x-pack/plugins/security_solution/public/management/components/console/components/command_input/lib/entered_input.tsx

```diff
@@ -100,36 +108,54 @@ export class EnteredInput {
           let argIndex = 0;
 
           // Loop through the input pieces (left and right side of cursor) looking for the Argument name
-          for (const { input, items } of inputPieces) {
+          for (const { input, items, side } of inputPieces) {
             const argNameMatch = `--${argName}`;
             let pos = input.indexOf(argNameMatch);
 
             while (pos > -1) {
               const argChrLength = argNameMatch.length;
-              const replaceValues: InputCharacter[] = Array.from(
-                { length: argChrLength },
-                createInputCharacter
-              );
-              const argState = enteredCommand.argState[argName]?.at(argIndex);
-
-              replaceValues[0] = createInputCharacter({
-                value: argNameMatch,
-                renderValue: (
-                  <ArgumentSelectorWrapper
-                    argName={argName}
-                    argIndex={argIndex}
-                    argDefinition={argDef as ArgumentSelectorWrapperProps['argDefinition']}
-                  />
-                ),
-                isArgSelector: true,
-                argName,
-                argIndex: argIndex++,
-                argState,
-              });
-
-              items.splice(pos, argChrLength, ...replaceValues);
-
-              pos = input.indexOf(argNameMatch, pos + argChrLength);
+              const startSearchIndexForNextArg = pos + argChrLength;
+              const charAfterArgName = input.charAt(startSearchIndexForNextArg);
+
+              // We need to Ensure that the argument name is not part of a longer word
+              // (example: matches `--file` not `--files`)
+              if (
+                pos > -1 &&
+                ARGUMENT_NAME_VALUE_SEPARATORS.includes(charAfterArgName) &&
+                ((side === 'left' &&
+                  charAfterArgName === '' &&
+                  // if cursor is at the end of the argument name on the left side of the cursor,
+                  // then we check to ensure that the first character of the Right side of the
+                  // cursor is one of the separators. Example (| === cursor): `--file|s`
+                  ARGUMENT_NAME_VALUE_SEPARATORS.includes(rightOfCursorText.charAt(0))) ||
+                  (side === 'left' && charAfterArgName !== '') ||
+                  side === 'right')
```

Comment:
This entire block of logic is becoming a bit hard to read and follow, especially with deep nested loops and conditionals. Is there a way we can simplify this? I'm concerned that going forward we might find it hard to debug and add changes here.

Reply:
I lifted some of this to a `var` so that this `if()` looks a bit easier to read